gnucash/gnome/gnc-plugin-page-invoice.c

```diff
@@ -421,6 +467,30 @@ gnc_plugin_page_invoice_update_menus (GncPluginPage *page, gboolean is_posted, g
                                "sensitive", can_unpost);
     gnc_plugin_update_actions (action_group, invoice_book_readwrite_actions,
                                "sensitive", !is_readonly);
+
+
+    for (i = 0; i < parametrized_n_actions; i++)
+    {
+        const gchar *action_name = parametrized_actions[i];
+        // need action by name, check whether in plugin page actions
+        for (j = 0; j < gnc_plugin_page_invoice_n_actions; j++)
+        {
+            action = gnc_plugin_page_invoice_actions[j];
+            if (action_name == action.name)
+            {
+                // found the action, update label and tooltip
+                const gchar *parametrized_label = action.label;
+                const gchar *parametrized_tooltip = action.tooltip;
+                GtkAction *gtk_action = gtk_action_group_get_action (action_group, action_name);
+                gtk_action_set_label(gtk_action,
+                                    g_strdup_printf ( _(parametrized_label),
+                                                    invoice_type_name));
```

Comment:
I'm pretty sure this leaks memory. A GtkAction takes a `const gchar*`, which typically means it internally makes a copy of the string and won't further touch the pointer passed in. However g_strdup_printf allocates a new string and this string is never freed.

Reply:
I was aware it could be leaking, but I saw things similar to this elsewhere in the gnucash code, and I hoped (rather than assumed) that somehow the freeing would be done on update automatically. The rewrite to static strings made the use of `g_strdup_printf` no longer necessary.